### PR TITLE
ERM-3163 On attempting to save an external data source if the save fa…

### PR DESCRIPTION
**Pull request by CalamityC**
…ils the toast message should reflect this

* adjust error handling in ExternalDataSourcesSettingsRoute
* reset form after edit submit because in case of an error dirty field stays in form state

### CODE REVIEW

File: src/settings/routes/ExternalDataSourcesSettingsRoute/ExternalDataSourcesSettingsRoute.js [hunk 1]
```diff
@@ -50,10 +50,15 @@ const ExternalDataSourcesSettingsRoute = () => {
 
   const { mutateAsync: putExternalKB } = useMutation(
     ['ERM', 'KBs', 'PUT'],
-    (payload) => {
-      ky.put(`${KBS_ENDPOINT}/${payload.id}`, { json: payload }).json().then(() => {
+    async (payload) => {
+      try {
+        await ky.put(`${KBS_ENDPOINT}/${payload.id}`, { json: payload }).json();
```
Comment (EthanFreestone) [inline src/settings/routes/ExternalDataSourcesSettingsRoute/ExternalDataSourcesSettingsRoute.js:55]: Can this please be changed back to `payload => ky.put(...).then()` but then including a `.catch()` block for the caught stuff?

File: src/settings/components/ExternalDataSourcesConfig/ExternalDataSourcesView/ExternalDataSourcesView.js [hunk 2]
```diff
@@ -288,8 +288,9 @@ const ExternalDataSourcesView = ({
         }}
         mutators={{ ...arrayMutators }}
         onDelete={onDelete}
-        onSubmit={values => {
+        onSubmit={(values, form) => {
           onSubmit(values);
+          form.reset();
```
Comment (EthanFreestone) [inline src/settings/components/ExternalDataSourcesConfig/ExternalDataSourcesView/ExternalDataSourcesView.js:293]: I'm not sure about this... Does ensuring the initialValues state up above is correct not fix the issue?


If not, then I accept this solution, but would like a comment explaining why it's needed, and perhaps an issue in the stripes-kint-components gitlab for me to dig into what happens on FormModal submit. I'd rather not have to repeat this fix everywhere we use formModal

**Review by EthanFreestone — CHANGES_REQUESTED** [review]

File: src/settings/components/ExternalDataSourcesConfig/ExternalDataSourcesView/ExternalDataSourcesView.js (hunk 2 above)
Comment (CalamityC) [reply src/settings/components/ExternalDataSourcesConfig/ExternalDataSourcesView/ExternalDataSourcesView.js:293]: > I'm not sure about this... Does ensuring the initialValues state up above is correct not fix the issue?
No. The initialValues contain the correct data, but the field is still filled with the data from previous create resp. edit (when edit was not successful)

File: src/settings/routes/ExternalDataSourcesSettingsRoute/ExternalDataSourcesSettingsRoute.js (hunk 1 above)
Comment (CalamityC) [reply src/settings/routes/ExternalDataSourcesSettingsRoute/ExternalDataSourcesSettingsRoute.js:55]: See our slack conversation, will not change that

**Pull request merged**